### PR TITLE
Tools: autotest: add test for copter's NAV_DELAY function

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -1244,6 +1244,57 @@ def fly_vision_position(self):
         if ex is not None:
             raise ex
 
+    def fly_nav_delay(self):
+        '''fly a simple mission that has a delay in it'''
```
Comment (khancyr) [inline Tools/autotest/arducopter.py]: """

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -1244,6 +1244,57 @@ def fly_vision_position(self):
...
+
+        num_wp = self.load_mission("copter_nav_delay.txt")
+
+        self.mavproxy.send('mode loiter\n')  # stabilize mode
```
Comment (khancyr) [inline Tools/autotest/arducopter.py]: wrong comment

Comment (khancyr) [reply Tools/autotest/arducopter.py]: better remove it

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -1244,6 +1244,57 @@ def fly_vision_position(self):
...
+        self.wait_mode('LOITER')
+        self.progress("Waiting reading for arm")
+        self.wait_ready_to_arm()
+
+        self.context_push();
+
+        ex = None
+        try:
+            self.arm_vehicle()
+            self.mavproxy.send('mode auto\n')  # stabilize mode
```
Comment (khancyr) [inline Tools/autotest/arducopter.py]: remove comment

File: Tools/autotest/arducopter.py [hunk 4]
```diff
@@ -1244,6 +1244,57 @@ def fly_vision_position(self):
...
+            self.progress("Stopped for %u seconds (want >=%u seconds)" %
+                          (calculated_delay, want_delay))
+            if calculated_delay < want_delay:
+                raise NotAchievedException()
+
+        except Exception as e:
+            self.progress("Exception caught")
+            ex = e
+
+        self.set_rc(3, 1000)
```
Comment (khancyr) [inline Tools/autotest/arducopter.py]: need land no ?

File: Tools/autotest/arducopter.py (hunk 1 above)
Comment (peterbarker) [reply Tools/autotest/arducopter.py]: Why does it matter?

Comment (khancyr) [reply Tools/autotest/arducopter.py]: For autodoc... that we don't support yet. And to be compliant with pep8 style

File: Tools/autotest/arducopter.py (hunk 4 above)
Comment (peterbarker) [reply Tools/autotest/arducopter.py]: There's an RTL at the end of the mission.  We should probably consider what sorts of resets to do after an exception, 'though...

**Pull request merged**